utils/splunk/splunk_telemetry_base.py

```diff
@@ -71,7 +71,17 @@ def _dispatch_and_await_search(self, instance, saved_searches):
 
         for saved_search in saved_searches:
             try:
+                persist_status_key = instance.instance_config.base_url + saved_search.name
+                if self.status.data.get(persist_status_key) is not None:
+                    for (sid, saved_search) in self.status.data[persist_status_key]:
+                        instance.splunkHelper.finalize_sid(sid, saved_search)
+                    self.status.data[persist_status_key] = []
+                    self.status.persist(self.persistence_check_name)
                 sid = self._dispatch_saved_search(instance, saved_search)
+                if self.status.data.get(persist_status_key) is None:
```

Comment:
those 2 lines (81-82) are really needed? 
before the `_dispatch_saved_search` call we created the array for `data[persist_status_key]`